### PR TITLE
Force to add reference to System.ServiceModel.Primitives.dll

### PR DESCRIPTION
**Pull request by huanwu**
Force to add reference to System.ServiceModel.Primitives.dll through new defined smreference parameter to avoid mismatched SM assembly error. 

#3004 
@mconnew @Lxiamail @yujayee

### CODE REVIEW

File: src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/Options.cs [hunk 1]
```diff
@@ -291,6 +292,26 @@ private void ParseReferenceAssemblies()
                 _parent._typeResolver = CreateTypeResolver(_parent);
             }
 
+            private void ParseSMReferenceAssembly()
+            {
+                IList<string> referencedAssembliesArgs = _arguments.GetArguments(Options.Cmd.SMReference);
+                if(referencedAssembliesArgs!= null && referencedAssembliesArgs.Count > 0)
+                {
```
Comment (mconnew) [inline src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/Options.cs]: Fix formatting

File: src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/Options.cs [hunk 2]
```diff
@@ -365,8 +386,6 @@ private void LoadReferencedAssemblies(IList<string> referenceArgs)
                     {
                         if (Tool.IsFatal(e))
                             throw;
-
-                        throw new ToolOptionException(SR.Format(SR.ErrCouldNotLoadReferenceAssemblyAt, path), e);
                     }
```
Comment (mconnew) [inline src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/Options.cs]: We should emit a warning message saying we were unable to load this reference assembly.

**Comment by StephenBonikowsky** [comment]
@huanwu is this PR still needed?

**Comment by huanwu** [comment]
@StephenBonikowsky no, I will close it.

**Pull request closed without merging**